File: 452-minNumArrowBurstBallon/solution.java
class Solution {
    public int findMinArrowShots(int[][] points) {

        // 4%
        Arrays.sort(points, (a, b) -> 
                    (a[0] == b[0] ? /*b.start - a.start*/1 : b[0]-a[0] ));
        // sorted by end (ascend)
        long start = Long.MAX_VALUE;
        int count = 0;
        for (int i = 0; i < points.length; i++) {
            if (points[i][1] < start){
                start = points[i][0];
                count++;
            }             
        }
        return count;


        // 10%
        if(points.length<2) return points.length

        Arrays.sort(points, (a, b) -> (b[0]-a[0]));
        // sorted by end (ascend)
        int start = points[0][0];
        int count = 1;
        for (int i = 1; i < points.length; i++) {
            if (points[i][1] < start){
                start = points[i][0];
                count++;
            }             
        }
        return count;

    }
}

File: 452-minNumArrowBurstBallon/solution.py
class Solution(object):
    def findMinArrowShots(self, points):
        """
        :type points: List[List[int]]
        :rtype: int
        """
        counter=0
        if not points:
            return counter

        length = len(input)
        '''
        # greedy, sort & check startInd O(N^log(N)), 83%
        input=sorted(points ,key=lambda l:l[0], reverse=True)
        previous_start=float('inf')

        for i in range(0, length):
            if previous_start>input[i][1]:
                previous_start=input[i][0]
                counter+=1

        return counter
        '''

        # greedy, check endInd O(N^log(N)), 70%
        points=sorted(points ,key=lambda l:l[1])
        previous_end=float('-inf')

        for point in points:
            if previous_end<point[i][0]:
                previous_end=point[i][1]
                counter+=1

        return counter
